MetaMask/metamask-extension
# Extract "create vault" form to separate component

The form used for creating a vault on the "Import" page of onboarding and on the "Restore vault" page is nearly identical, yet the implementation is totally separate. It has now been extracted to a separate component, consolidating the two implementations.

There is a "terms of use" checkbox on the import page that isn't on the restore vault page, so that part has been made optional. The "submit" button text differs between the two uses as well, so that is customizable.

There are slight styling differences between the old and new versions of this form. The fonts and spacing are all using our new standard design system guidelines, and we're using our standard checkbox now as well. The spacing and font sizes were chosen somewhat arbitrarily by me to resemble the old styles, so please feel free to suggest changes if you think they can be improved upon.

There are some slight copy changes to the "Restore vault" page as well; the placeholder text and the label for the "Secret Recovery Phrase" field now matches the "Import" page copy.

Manual testing steps:  
  - Import a SRP during onboarding and ensure it works correctly and looks OK
  - Lock the wallet, then go through the "Forgot password" flow to restore the SRP. Ensure it works correctly and looks OK.

<details>
<summary>Screenshots:</summary>

## Before

Import screen (SRP hidden)

![current-import-hidden](https://user-images.githubusercontent.com/2459287/151859715-177e4ec2-b8bc-4b56-bda0-f9101e011c9b.png)

Import screen (SRP shown)

![current-import-shown](https://user-images.githubusercontent.com/2459287/151859709-30c35518-ff43-48a8-8f24-80597c0e7e64.png)

Restore screen (SRP hidden)

![current-restore-hidden](https://user-images.githubusercontent.com/2459287/151859741-6528c999-c853-45a1-80ac-64b19a2b3987.png)

Restore screen (SRP shown)

![current-restore-shown](https://user-images.githubusercontent.com/2459287/151859736-2c92be4f-5e6f-459a-aa69-90f6f0300b44.png)

## After

Import screen (SRP hidden)

![new-import-hidden](https://user-images.githubusercontent.com/2459287/151859782-2f121582-8fcd-4971-a363-d92e55b7e6e9.png)

Import screen (SRP shown)

![new-import-shown](https://user-images.githubusercontent.com/2459287/151859776-d188b7e8-ea4d-41d7-806c-c5d97ca35c30.png)

Restore screen (SRP hidden)

![new-restore-hidden](https://user-images.githubusercontent.com/2459287/151859761-1ef2ffa6-de93-4e49-ae30-02ef04b46c55.png)

Restore screen (SRP shown)

![new-restore-shown](https://user-images.githubusercontent.com/2459287/151859757-61271c5d-2f1b-43d0-bff9-492645f45747.png)

</details>
